app/src/main/java/org/simple/clinic/selectstate/SelectStateEffectHandler.kt

```diff
@@ -32,7 +32,8 @@ class SelectStateEffectHandler @AssistedInject constructor(
     return ObservableTransformer { effects ->
       effects
           .observeOn(schedulers.io())
-          .map { appConfigRepository.saveState(it.state) }
+          .doOnNext { appConfigRepository.saveState(it.state) }
+          .doOnNext { appConfigRepository.saveDeployment(it.state.deployment) }
```

Comment:
Why do we save deployments? Is that for our reference or something else that I'm missing?

Reply:
Deployments are different from the state, for example, each deployment can have multiple states. So instead of tying a state and deployment together, we are saving them separately.